test/replica-isolation/mzcompose.py

```diff
@@ -32,6 +34,135 @@
 ]
 
 
+class CompactionCheck:
+    # replica: a string describing the SQL accessible name of the replica. Example"cluster1.replica1"
+    # host: docker container name from which to check the log. Example: "computed_1_1"
+    def __init__(self, replica: str, host: str):
+        assert "." in replica
+        self.replica = replica
+        self.host = host
+        self.id: Optional[str] = None
+        self.satisfied = False
+
+    def find_id(self, c: Composition) -> None:
+        assert False
+
+    def print_error(self) -> None:
+        assert False
+
+    def check_log(self, c: Composition) -> None:
+        self.find_id(c)
+        assert self.id is not None
+        log: str = c.invoke("logs", self.host, capture=True).stdout
+        for line in [x for x in log.splitlines() if x.find("ClusterClient send=") > -1]:
+            if "AllowCompaction" in line:
+                if self.id in line:
+                    self.satisfied = True
+        if not self.satisfied:
+            self.print_error()
```

Comment:
You can just assert here unless you want to accumulate the results of all the checks.